GooglePlayInstant/Tests/Editor/AndroidManifest/AndroidManifestHelperTests.cs

```diff
@@ -0,0 +1,272 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Xml.Linq;
+using GooglePlayInstant.Editor.AndroidManifest;
+using NUnit.Framework;
+
+namespace GooglePlayInstant.Tests.Editor.AndroidManifest
+{
+    [TestFixture]
+    public class AndroidManifestHelperTests
+    {
+        // The following constants are duplicated between AndroidManifestHelper and here since they are part of the
+        // Android manifest schema. Changing one of these strings in AndroidManifestHelper should break a test.
+        private const string Action = "action";
```

Comment:
some of these are dupes, should this go into a dedicated Constants class instead?

Reply:
I did this intentionally for the reason described in the comment above. It's sort of a philosophical position. Note that I only copied string constants that are part of the AndroidManifest schema. I didn't copy other constants, e.g. the "Precondition" error messages.

Let me know if you're strongly opposed to this philosophy.